### PR TITLE
Improve visibility of startup problems

### LINKED ISSUE #1873: Improve visibility of plugin startup problems
If the plugin detects a fatal error condition and exits, it gets restarted by Docker because of the `--restart=always`.  However this situation is not made very clear to the user.

For example, a bad option will cause an immediate exit:

````
$ weave launch-plugin --foo
7c4094d284ecec15addf8a2e800cc42a2509db51f4675e5a7479eeae45a03731
````

No hint here, and worse, `weave status` gives a normal report:

````
$ weave status
[...]
        Service: plugin
     DriverName: weave
````

Compare:

````
$ weave launch-router --foo
The weave container has died. Consult the container logs for further details.
````

Note #1814 fixed some related things in the weave script.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #1873

Also improves behaviour in the case that the user has set a restart policy on weaver or weaveproxy, and then encountered a problem.

Replacement for #1879, now on master branch

### CODE REVIEW

File: net/listen.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import (
+	"fmt"
+	"net"
+	"os"
+)
+
+func ListenUnixSocket(pathname string) (net.Listener, error) {
+	if err := os.Remove(pathname); err != nil && !os.IsNotExist(err) {
+		return nil, fmt.Errorf("Error removing existing status socket: %s", err)
```
Comment (rade) [inline net/listen.go]: s/status/unix

**Comment by bboreham** [comment]
Fixed cutpasta

File: weave [hunk 2]
```diff
@@ -1286,6 +1280,7 @@ collect_dns_add_remove_args() {
     [ $# -eq 0 -a \( -n "$C" -o \( $IP_COUNT -gt 0 -a -n "$FQDN" \) \) ] || usage
     check_running $CONTAINER_NAME
     if [ -n "$C" ] ; then
+        check_running $C
```
Comment (rade) [inline weave]: why?

**Comment by rade** [comment]
This surprised me...

```
$ weave launch-router
dc4d47c5899bb063a771415f8290f83c87bfb19de6f82326370cae5e5c141c69
$ weave launch-plugin --foo
The weaveplugin container has died. Consult the container logs for further details.
$ weave launch-plugin
weaveplugin is restarting; you can stop it with 'weave stop-plugin'.
```
Perhaps we should do a `docker kill` in `wait_for_status`?

File: weave (hunk 2 above)
Comment (bboreham) [inline weave]: to maintain the previous behaviour; `container_id` used to do this check

**Comment by bboreham** [comment]
> Perhaps we should do a docker kill in wait_for_status?

Done.

File: weave (hunk 2 above)
Comment (rade) [inline weave]: `container_id` is called from lots of places. Yet you only changed this one. What makes this one special?

Comment (bboreham) [inline weave]: all the others are covered by nearby checks

**Comment by rade** [comment]
This works well. Not just for the plugin, but e.g.
```
WEAVE_DOCKER_ARGS="--restart=always" weave launch --foo
```
will now error and stop the container instead of hanging.

**Pull request merged**